tools/windows/DatadogAgentInstaller/CustomActions/Telemetry.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using Microsoft.Deployment.WindowsInstaller;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+
+namespace Datadog.CustomActions
+{
+    public class Telemetry
+    {
+        private readonly IInstallerHttpClient _client;
+        private readonly ISession _session;
+
+        public const string DefaultSite = "datadoghq.com";
+
+        public Telemetry(
+            IInstallerHttpClient client,
+            ISession session)
+        {
+            _client = client;
+            _session = session;
+        }
+
+        public static Telemetry DefaultTelemetry(ISession session)
+        {
+            return new Telemetry(new InstallerWebClient(), session);
+        }
+
+        public void ReportTelemetry(string eventName)
+        {
+            var apikey = _session["APIKEY"];
+            var site = _session["SITE"];
+            if (string.IsNullOrEmpty(site))
+            {
+                site = DefaultSite;
+            }
+            var uri = $"https://instrumentation-telemetry-intake.{site}/api/v2/apmtelemetry";
+
+            var installerVersion = Assembly.GetExecutingAssembly().GetName().Version;
+            var agentVersion = CiInfo.PackageVersion;
+            var payload = @$"
+{{
+    ""request_type"": ""apm-onboarding-event"",
+    ""api_version"": ""v1"",
+    ""payload"": {{
+        ""event_name"": ""{eventName}"",
+        ""tags"": {{
+            ""agent_platform"": ""Windows"",
+            ""agent_version"": ""{ agentVersion}"",
+            ""script_version"": ""{ installerVersion}""
+        }}
+    }}
+}}";
```

Comment:
Where did you get this format from. And it is interesting you type is apm-onboarding-event. Is this what our Linux installation is doing now?

Reply:
Just strange that it is called apm-onboarding-event considering what we do here